disperser/batcher/txn_manager.go

```diff
@@ -0,0 +1,160 @@
+// It returns an error if the transaction fails to be sent for reasons other than timeouts.
+// TxnManager monitors the transaction and resends it with a higher gas price if it is not mined without a timeout.
+func (t *TxnManager) ProcessTransaction(ctx context.Context, req *TxnRequest) error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	gasTipCap, gasFeeCap, err := t.ethClient.GetLatestGasCaps(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get latest gas caps: %w", err)
+	}
+
+	txn, err := t.ethClient.UpdateGas(ctx, req.Tx, req.Value, gasTipCap, gasFeeCap)
+	if err != nil {
+		return fmt.Errorf("failed to update gas price: %w", err)
+	}
+	err = t.ethClient.SendTransaction(ctx, txn)
+	if err != nil {
+		return fmt.Errorf("failed to send txn (%s) %s: %w", req.Tag, req.Tx.Hash().Hex(), err)
+	}
+
+	t.requestChan <- req
+	return nil
+}
+
+// monitorTransaction monitors the transaction and resends it with a higher gas price if it is not mined without a timeout.
+// It returns an error if the transaction fails to be sent for reasons other than timeouts.
+func (t *TxnManager) monitorTransaction(ctx context.Context, req *TxnRequest) (*types.Receipt, error) {
+	for {
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, t.txnRefreshInterval)
+		defer cancel()
+
+		receipt, err := t.ethClient.EnsureTransactionEvaled(
+			ctxWithTimeout,
+			req.Tx,
+			req.Tag,
+		)
+		if err == nil {
+			return receipt, nil
+		}
+
+		if errors.Is(err, context.DeadlineExceeded) {
+			t.logger.Warn("transaction not mined within timeout, resending with higher gas price", "tag", req.Tag, "txHash", req.Tx.Hash().Hex())
+			txn, err := t.speedUpTxn(ctxWithTimeout, req.Tx, req.Tag)
+			if err != nil {
+				t.logger.Error("failed to speed up transaction", "err", err)
+				return nil, err
+			}
+			err = t.ethClient.SendTransaction(ctx, txn)
+			if err != nil {
+				t.logger.Error("failed to send txn", "tag", req.Tag, "txn", req.Tx.Hash().Hex(), "err", err)
+				return nil, err
+			}
+		} else {
+			t.logger.Error("transaction failed", "tag", req.Tag, "txHash", req.Tx.Hash().Hex(), "err", err)
+			return nil, err
+		}
+	}
+}
+
+// speedUpTxn increases the gas price of the existing transaction by specified percentage.
+// It makes sure the new gas price is not lower than the current gas price.
+func (t *TxnManager) speedUpTxn(ctx context.Context, tx *types.Transaction, tag string) (*types.Transaction, error) {
+	prevGasTipCap := tx.GasTipCap()
+	prevGasFeeCap := tx.GasFeeCap()
+	// get the gas tip cap and gas fee cap based on current network condition
+	currentGasTipCap, currentGasFeeCap, err := t.ethClient.GetLatestGasCaps(ctx)
+	if err != nil {
+		return nil, err
+	}
+	increasedGasTipCap := increaseGasPrice(prevGasTipCap)
+	increasedGasFeeCap := increaseGasPrice(prevGasFeeCap)
+	// make sure increased gas prices are not lower than current gas prices
+	var newGasTipCap, newGasFeeCap *big.Int
+	if currentGasTipCap.Cmp(increasedGasTipCap) > 0 {
```

Comment:
Should we compare to `currentGasTipCap` or `increaseGasPrice(currentGasTip)`?

Reply:
Not sure what you meant here. `currentGasTipCap` is being compared to `increasedGasTipCap`

Reply:
I was proposing something like this: 

```
if increasedGastTipPrice < increaseGasPrice(currentGasTip){
    newGasTip =  increaseGasPrice(currentGasTip
}
```

I was just wondering about a (possibly unlikely) scenario where gas prices are continually increasing, so all we are doing is keeping up with gas prices instead of ever getting our transaction some priority.

Reply:
`currentGasTipCap` is the gas tip cap according to current network condition (25% more than what's suggested from execution client). So it's comparing the gas tip cap according to the current network condition (`currentGasTipCap`) and the gas tip cap boosted from the previously specified gas tip cap. 
If we use `increaseGasPrice(currentGasTipCap)`, it would be adding 10% in addition to 25% extra added to what's being suggested from execution client.